### DISCUSSION TITLE
[Begginer] How to preserve search params when navigating to child routes

### DISCUSSION THREAD

**Discussion by Trictonicmp** [TanStack/router#2289] [Q&A]
Hello everyone, this may be a silly question with a simple answer but I haven't figured out yet so I ask for help here.

I have a route like this:
```ts
const shirtsSectionRouteDefinition = createRoute({
  getParentRoute: () => inventoryRoute,
  path: 'shirts',
  loader: () => appQueryClient.ensureQueryData(shirtsQueryOptions),
  validateSearch: (search) => shirtsSectionSearchSchema.parse(search)
}).lazy(() =>
  import("./ShirtsSection-route.lazy.module").then(
    (d) => d.ShirtsSectionLazyRoute,
  ),
);

export const shirtsSectionRoute = shirtsSectionRouteDefinition.addChildren([
  shirtMainDetailsProxyRoute,
  addShirtFromSectionRoute,
]);
```

and within this route's component I have an outlet for a modal, that I use to display `shirtMainDetailsProxyRoute` or `addShirtFromSectionRoute`, but when I go to any of these routes, search params get cleared, I have tried different approaches like:

```ts
const openAddShirt = () => {
     router.navigate({
      from: shirtSectionRoute.fullPath,
      to: addShirtFromSectionRoute.to
    });
  };
```
and also tried with
```ts
const openAddShirt = () => {
     router.navigate({
      from: router.history.location.href,
      to: addShirtFromSectionRoute.to
    });
  };
```
but this last just broke the routing as it appends search params at the end.

this also makes it harder to go close the modal as it erases all of the params, making it harder to deal with a params state.

I hope I explained myself.

**Comment by Trictonicmp** [comment]
NVM, I realized taht I can pass search params to all child route, quite annoying but it works, It would be great to have something similar to "relativeTo" in Angular's router. 
So here is the solution to what I was asking here:
```ts
const openAddShirt = () => {
     router.navigate({
      from: router.history.location.href,
      to: addShirtFromSectionRoute.to,
      search: (prev) => ({...prev}),
    });
  };
```

**Reply by schiller-manuel** [reply]
you can also set search to `true` for the same effect.
we could think of copying over search params when using relative routes, but this is likely a breaking change so we would need to be careful.

btw `from` should only be a valid route not something taken from the location

**Reply by Trictonicmp** [reply]
Didn't know this, thanks for sharing!
